### PR TITLE
Fill device state

### PR DESCRIPTION
**Pull request by jgrund**
The device plugin emits a value upon connecting, but does not fill the
internal state with this value.

Because of this, data will always be sent for the first two poll
intervals, even if nothing has changed.

Always fill the state, even when resolving the initial future.

Signed-off-by: Joe Grund <jgrund@whamcloud.io>

---
This change is [Reviewable](https://reviewable.io/reviews/whamcloud/integrated-manager-for-lustre/1869)

### CODE REVIEW

**Comment by mkpankov** [comment]
I'm still seeing 2 initial updates, did you observe that as well?

**Comment by mkpankov** [comment]
What I was doing specifically: run `import-pools`, restart the `iml-manager`. Then there are two updates on `iml-device` side. This happens every restart of the manager.

**Comment by jgrund** [comment]
> I'm still seeing 2 initial updates, did you observe that as well?

Good catch, I should start the state as Send instead of pending.

File: iml-agent/src/daemon_plugins/device.rs [hunk 1]
```diff
@@ -31,17 +30,23 @@ fn device_stream() -> impl Stream<Item = Result<String, ImlAgentError>> {
...
+enum State {
+    Pending,
+    Sent,
+}
+
 pub fn create() -> impl DaemonPlugin {
     Devices {
         trigger: None,
-        state: Arc::new(Mutex::new((None, None))),
+        state: Arc::new(Mutex::new((None, State::Pending))),
```
Comment (jgrund) [inline iml-agent/src/daemon_plugins/device.rs:42]:
```suggestion
        state: Arc::new(Mutex::new((None, State:: Sent))),
```

**Comment by jgrund** [comment]
> While it fixed the issue, it looks counter-intuitive to start in Sent state. Maybe call it Ready or something?

Yeah, it is a little counter-intuitive.

State is currently:

```rust
#[derive(Eq, PartialEq)]
enum State {
    Pending,
    Sent,
}
```

`Sent` basically means the value has been sent. This will be the case if the start_session method succeeds. 

`Pending` means there is a new value to be sent by `update_session`. 

If the `start_session` method fails, then we never call `update_session` any way, so it doesn't make sense to have another state here.

**Pull request merged**